src/commons/editor/EditorContainer.tsx

```diff
@@ -1,24 +1,61 @@
+// Necessary to prevent "ReferenceError: ace is not defined" error.
+// See https://github.com/securingsincity/react-ace/issues/1233 (although there is no explanation).
+import 'ace-builds/src-noconflict/ace';
+
 import _ from 'lodash';
 import React from 'react';
 
+import SourcecastEditor, {
+  SourceRecorderEditorProps
+} from '../sourceRecorder/SourceRecorderEditor';
 import { EditorTabState } from '../workspace/WorkspaceTypes';
 import Editor, { EditorProps, EditorTabStateProps } from './Editor';
 
-export type EditorContainerProps = Omit<EditorProps, keyof EditorTabStateProps> & {
+export type NormalEditorContainerProps = Omit<EditorProps, keyof EditorTabStateProps> & {
+  editorVariant: 'normal';
   editorTabs: EditorTabStateProps[];
 };
 
+export type SourcecastEditorContainerProps = Omit<
+  SourceRecorderEditorProps,
+  keyof EditorTabStateProps
+> & {
+  editorVariant: 'sourcecast';
+  editorTabs: EditorTabStateProps[];
+};
+
+export type EditorContainerProps = NormalEditorContainerProps | SourcecastEditorContainerProps;
+
 export const convertEditorTabStateToProps = (editorTab: EditorTabState): EditorTabStateProps => {
   return {
     editorValue: editorTab.value,
     ..._.pick(editorTab, 'highlightedLines', 'breakpoints', 'newCursorPosition')
   };
 };
 
+const createNormalEditorTab =
+  (editorProps: Omit<EditorProps, keyof EditorTabStateProps>) =>
+  (editorTabStateProps: EditorTabStateProps) => {
+    return <Editor {...editorProps} {...editorTabStateProps} />;
+  };
+
+const createSourcecastEditorTab =
+  (editorProps: Omit<SourceRecorderEditorProps, keyof EditorTabStateProps>) =>
+  (editorTabStateProps: EditorTabStateProps) => {
+    return <SourcecastEditor {...editorProps} {...editorTabStateProps} />;
+  };
+
 const EditorContainer: React.FC<EditorContainerProps> = (props: EditorContainerProps) => {
-  const { editorTabs, ...editorProps } = props;
+  let createEditorTab;
+  if (props.editorVariant === 'sourcecast') {
+    const { editorVariant, editorTabs, ...editorProps } = props;
+    createEditorTab = createSourcecastEditorTab(editorProps);
+  } else {
+    const { editorVariant, editorTabs, ...editorProps } = props;
+    createEditorTab = createNormalEditorTab(editorProps);
+  }
```

Comment:
The code is written as such so that TypeScript can automatically infer and narrow the types for us. I've tried to write it more succinctly, but TypeScript was unable to narrow the types automatically (meaning that type casting would be needed). If you can find a more succinct way of writing this, do let me know.